flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/CastRuleUtils.java

```diff
@@ -233,11 +233,39 @@ public CodeWriter ifStmt(
             return this;
         }
 
+        public CodeWriter tryCatchStmt(
+                Consumer<CodeWriter> bodyWriterConsumer,
+                Class<? extends Throwable> catchClass,
```

Comment:
Maybe you can have another method without `catchClass` arg which just calls this with `Throwable.class`, as I'm guessing that's the most common usage.